roxie/ccd/ccdserver.cpp

```diff
@@ -19358,10 +19369,10 @@ class CRoxieServerWhenActivity : public CRoxieServerActivity
         }
         else if (state != STATEstopped)
         {
-            stopDependencies(savedExtractSize, savedExtract, aborting ? WhenSuccessId : WhenFailureId);  // These ones don't get executed
-            executeDependencies(savedExtractSize, savedExtract, aborting ? WhenFailureId : WhenSuccessId); // These ones do
+            stopDependencies(savedExtractSize, savedExtract, aborted ? WhenSuccessId : WhenFailureId);  // These ones don't get executed
+            executeDependencies(savedExtractSize, savedExtract, aborted ? WhenFailureId : WhenSuccessId); // These ones do
```

Comment:
alternatively the graph::abort code should set aborted on all activities not just the sinks